server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -339,7 +338,9 @@ public Void run() {
      * @return a warning value formatted according to RFC 7234
      */
     public static String formatWarning(final String s) {
-        return String.format(Locale.ROOT, WARNING_FORMAT, escapeAndEncode(s), RFC_7231_DATE_TIME.format(ZonedDateTime.now(GMT)));
+        return WARNING_PREFIX + " "
+                + "\"" + escapeAndEncode(s) + "\"" + " "
+                + "\"" + STARTUP_TIME + "\"";
```

Comment:
Yes, what is here is clearer IMO, and the compiler will fold the constants anyway (so fold `" "` and `"\""` into `" \""`, and `"\""` and `" "` and `"\"` into `"\" \""` so that the number of invocations to `StringBuilder#append` will be reduced).

Reply:
Fine with me.